dali/python/nvidia/dali/_debug_mode.py

```diff
@@ -419,57 +442,111 @@ def _external_source(self, name=None, **kwargs):
             raise RuntimeError(f"Unexpected operator 'ExternalSource'. Debug mode does not support"
                                " changing the order of operators executed within the pipeline.")
 
-    def _run_subpipeline(self, pipe_tuple, op_name, inputs, kwargs):
-        """Run pipeline wrapper of a single operator."""
-        self._debug_on = False
-        pipe, inputs_es, kwargs_es = pipe_tuple
+    def _run_op_on_device(self, op_name, device, inputs, kwargs):
```

Comment:
Maybe I have missed that, but how do we enforce the regular pipeline flow that GPU op cannot be followed by the CPU one, or we don't?

Reply:
Thanks for the catch, we didn't. Added check for that.

Reply:
Do you mean test itself or there is a check in the code added?
I was thinking about the following flow:
```
x = op1(devide="cpu")
y = op2(x.as_gpu(), devide="gpu")
z = op3(y.as_cpu(), devide="cpu")
```

Reply:
Agreed to stay with the current implementation (allowing to have CPU operator after a GPU one as long as the data is on correct device) to support external data (like `numpy.ndarray`) in debug operators.